### DISCUSSION TITLE
Docker build warning

### DISCUSSION THREAD

**Discussion by fhkong** [cmu-db/bustub#384] [Q&A]
I use docker to build my own build environment.
But when I type `cmake -DCMAKE_BUILD_TYPE=Debug ..`
The following message occurs
```
CMake Warning at CMakeLists.txt:41 (message):
  !! We recommend that you use clang-12 for developing BusTub.  You're using
  GNU 11.2.0, which is not clang.
```

**Comment by skyzh** [comment]
Read the content. Use the right compiler.

**Comment by q2333gh** [comment]
i searched the net ,some content might be helpful.
https://askubuntu.com/questions/1198087/how-to-set-clang-9-as-the-default-c-compiler-on-ubuntu-19-10
https://cmake.org/cmake/help/latest/command/add_compile_options.html
https://blog.csdn.net/rangfei/article/details/108862896

my goal is to  :
only set this project default compilier using clang. it effect all under the bustub folder.
im still solving this.

**Reply by llescoat** [reply]
Hi, have you figured out how to set the project's default compiler to `clang`, without modifying the default compiler for other projects?

Thanks.

**Reply by q2333gh** [reply]
maybe use docker is a option
